pygments/lexers/q.py

```diff
@@ -0,0 +1,235 @@
+                Number.Hex,
+            ),
+            # Byte Vectors
+            (r"0x[0-9a-fA-F]+", Number.Hex),
+            # Floats
+            (
+                r"([0-9]*[.]?[0-9]+|[0-9]+[.]?[0-9]*)[eE][+-]?[0-9]+[ef]?",
+                Number.Float,
+            ),
+            (r"([0-9]*[.][0-9]+|[0-9]+[.][0-9]*)[ef]?", Number.Float),
+            (r"[0-9]+[ef]", Number.Float),
+            # Characters
+            (r"[0-9]+[c]", Number),
+            # Integers
+            (r"[0-9]+[ihtuv]", Number.Integer),
+            # Long Integers
+            (r"[0-9]+[jnp]?", Number.Integer.Long),
+        ],
+        "comments": [
+            (r"[^\\]+", Comment.Multiline),
+            (r"^\\", Comment.Multiline, "#pop"),
+            (r"[\\]", Comment.Multiline),
+        ],
+        "strings": [
+            (r'[^"\\]+', String.Double),
+            (r"\\.", String.Escape),
+            (r'"', String.Double, "#pop"),
+        ],
+    }
+
+
+class QLexer(KLexer):
+    """
+    For `Q <https://code.kx.com/>`_ source code.
+
+    .. versionadded:: 2.12
+    """
+
+    name = "Q"
+    aliases = ["q"]
+    filenames = ["*.q"]
+
+    tokens = {
+        "root": [
+            (
+                words(
+                    ("aj", "aj0", "ajf", "ajf0", "all", "and", "any",
+                     "asc", "asof", "attr", "avgs", "ceiling", "cols",
+                     "count", "cross", "csv", "cut", "deltas", "desc",
+                     "differ", "distinct", "dsave", "each", "ej",
+                     "ema", "eval", "except", "fby", "fills", "first",
+                     "fkeys", "flip", "floor", "get", "group", "gtime",
+                     "hclose", "hcount", "hdel", "hsym", "iasc",
+                     "idesc", "ij", "ijf", "inter", "inv", "key",
+                     "keys", "lj", "ljf", "load", "lower", "lsq",
+                     "ltime", "ltrim", "mavg", "maxs", "mcount", "md5",
+                     "mdev", "med", "meta", "mins", "mmax", "mmin",
+                     "mmu", "mod", "msum", "neg", "next", "not",
+                     "null", "or", "over", "parse", "peach", "pj",
+                     "prds", "prior", "prev", "rand", "rank", "ratios",
+                     "raze", "read0", "read1", "reciprocal", "reval",
+                     "reverse", "rload", "rotate", "rsave", "rtrim",
+                     "save", "scan", "scov", "sdev", "set", "show",
+                     "signum", "ssr", "string", "sublist", "sums",
+                     "sv", "svar", "system", "tables", "til", "trim",
+                     "txf", "type", "uj", "ujf", "ungroup", "union",
+                     "upper", "upsert", "value", "view", "views", "vs",
+                     "where", "wj", "wj1", "ww", "xasc", "xbar",
+                     "xcol", "xcols", "xdesc", "xgroup", "xkey",
+                     "xlog", "xprev", "xrank"),
+                    suffix=r"\b",
+                ),
+                Operator.Word,
```

Comment:
`Name.Builtin`, not `Operator.Word`?

Reply:
my goal is to have the K (C++ defined operators) in one  style (i chose `Keyword`) and the Q operators (defined as macros on top of K) in another style.  I chose `Operator.Word`  because i usually refer to them as operators and the documentation made sense to me.  
```
Operator.Word
    For any operator that is a word (e.g. not).
```
in fact, each of the Q operators can be infix just like a single character operator.  in fact many of the names (such as ``reverse`` are just a single character in K ``|`` but given a name in Q to make the language more palatable.

i'm having trouble intuiting the difference between your suggestion ``Name.Builtin`` and ``Operator.Word``.  how do you typically differentiate the two? i've tested the output with with html and instead of highlighting in purple, ``Name.Builtin``  is green as well (just not bold like ``Keyword``.  happy to switch if it makes more sense to you.

Reply:
Typically, `Name.Builtin` is for built-in functions (think Python `print()`, `open()`, etc.) whereas `Operator.Word` is for infix operators that are not spelt with special characters but as a word, and usually are reserved names (can't be used as variables), which corresponds to Python's `and`, `or`, `not` and `is`. I hadn't realized these could be used with infix syntax. In that case, `Operator.Word` should be OK (it's not always possible to map concepts 1:1 between languages). By the way, note that the HTML output you tested uses the default style, there are many more. Have a look at https://pygments.org/styles/. There is no `Operator.Word` in the example, but if you want to see how it looks like, change `doc/examples/example.py` and build the docs locally.